src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala

```diff
@@ -150,10 +150,17 @@ object ConsignmentFields {
         "files",
         ListType(FileType),
         arguments = FileFiltersInputArg :: Nil,
-        resolve = ctx => {
+        resolve = Projector((ctx, projected) => {
           val fileFilters = ctx.args.argOpt("fileFiltersInput")
-          DeferFiles(ctx.value.consignmentid, fileFilters)
-        }
+          projected match {
+            case v: Vector[ProjectedName] if v.exists(_.name == "originalFilePath") =>
+              for {
+                files <- ctx.ctx.fileService.getFileMetadata(ctx.value.consignmentid, fileFilters)
+                filesWithRedacted <- ctx.ctx.fileService.updateOriginalFilePath(ctx.value.consignmentid, files)
+              } yield filesWithRedacted
+            case _ => DeferFiles(ctx.value.consignmentid, fileFilters)
```

Comment:
Just a suggestion, can we make it more generic? Maybe something like:

`  case class QueriedFileFields(originalFilePath: Boolean = false)`

and then set it in the `Files` field

 ```
      Field(
        "files",
        ListType(FileType),
        arguments = FileFiltersInputArg :: Nil,
        resolve = Projector((ctx, projected) => {
          val fileFilters = ctx.args.argOpt("fileFiltersInput")
          val queriedFileFields = QueriedFileFields(projected.exists(_.name == "originalFilePath"))
          DeferFiles(ctx.value.consignmentid, fileFilters, queriedFileFields)
        })
      ),
```
and now, we can use it in the `getMetadata()` method to check if the field is queried.

  ```
def getFileMetadata(consignmentId: UUID, fileFilters: Option[FileFilters] = None, queriedFileFields: QueriedFileFields): Future[List[File]] = {
    val filters = fileFilters.getOrElse(FileFilters())
    for {
      fileAndMetadataList <- fileRepository.getFiles(consignmentId, filters)
      ffidMetadataList <- ffidMetadataService.getFFIDMetadata(consignmentId)
      avList <- avMetadataService.getAntivirusMetadata(consignmentId)
      ffidStatus <- fileStatusService.getFileStatus(consignmentId)
      redactedFilePairs <- if (queriedFileFields.originalFilePath) fileRepository.getRedactedFilePairs(consignmentId) else Future(List())
    } yield fileAndMetadataList.toFiles(avList, ffidMetadataList, ffidStatus).toList
  }
```
The above code is from your previous code changes.

So, this way we can check other fields too (i.e `metadata, ffidMetadata, antivirusMetadata`). You don't have to do it now for other fields.

Sorry, you have changed a lot of code from the previous commit, now I'm asking to change it back to the way it was with a bit of enhancement.

Reply:
That's a really good idea. I've implemented this. As you say, it wasn't a lot to change it, I just put back some of the changes from a previous commit.